File: frontend/types/prompt.ts
/**
 * Base prompt interface
 */
export interface PromptBase {
  prompt_text: string;
  prompt_type?: string;
}

/**
 * Prompt creation interface
 */
export interface PromptCreate extends PromptBase {
  project_id: string;
}

/**
 * Prompt update interface
 */
export interface PromptUpdate {
  prompt_text?: string;
  generated_content?: string;
  prompt_type?: string;
  status?: string;
}

/**
 * Prompt interface as returned from API
 */
export interface Prompt extends PromptBase {
  id: string;
  project_id: string;
  user_id: string;
  generated_content?: string;
  status: string;
  created_at: string;
  updated_at: string;
}

/**
 * Prompt with versions
 */
export interface PromptWithVersions extends Prompt {
  versions: PromptVersion[];
}

/**
 * Prompt version interface
 */
export interface PromptVersion {
  id: string;
  prompt_id: string;
  version_number: number;
  content: string;
  created_at: string;
}

File: backend/app/models/prompt.py
"""Prompt model for the application."""
from typing import List, Optional
from sqlalchemy import Column, String, Text, DateTime, ForeignKey
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import relationship
from sqlalchemy.sql import func
import uuid

from app.database import Base


class Prompt(Base):
    """Prompt model."""
    
    __tablename__ = "prompts"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    project_id = Column(UUID(as_uuid=True), ForeignKey("projects.id", ondelete="CASCADE"), nullable=False)
    user_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False)
    prompt_text = Column(Text, nullable=False)
    generated_content = Column(Text, nullable=True)
    generated_at = Column(DateTime(timezone=True), nullable=True)
    prompt_type = Column(String(50), nullable=True)
    status = Column(String(50), default="pending")
    created_at = Column(DateTime(timezone=True), server_default=func.now())
    updated_at = Column(DateTime(timezone=True), server_default=func.now(), onupdate=func.now())
    
    # Relationships
    project = relationship("Project", back_populates="prompts")
    user = relationship("User", back_populates="prompts")
    versions = relationship("PromptVersion", back_populates="prompt", cascade="all, delete-orphan")
    
    def __repr__(self):
        return f"<Prompt {self.id}>"